main.go

```diff
@@ -120,3 +120,34 @@ func getEnvVars(vars map[string]string) (map[string]string, error) {
 	}
 	return result, nil
 }
+
+func prCheckEarlyExit(envVars map[string]string) {
+	// indicates PR check when set
+	prevBundleSha := os.Getenv("PREVIOUS_BUNDLE_SHA")
+	if len(prevBundleSha) > 0 {
+		prCheckGqlFile := os.Getenv("GRAPHQL_PRCHECK_QUERY_FILE")
+		if prCheckGqlFile == "" {
+			prCheckGqlFile = "/queries/prCheck.graphql"
+		}
+		saasName := os.Getenv("GIT_PARTITION_SAAS_NAME")
+		if saasName == "" {
+			saasName = "saas-git-partition-sync-producer"
+		}
+		ctx := context.Background()
+		canExit, err := pkg.EarlyExit(
+			ctx,
+			envVars["GRAPHQL_SERVER"],
+			prCheckGqlFile,
+			envVars["GRAPHQL_USERNAME"],
+			envVars["GRAPHQL_PASSWORD"],
+			prevBundleSha,
+			saasName,
+		)
+		if err != nil {
+			log.Fatalln(err)
+		} else if canExit {
+			log.Println("Relevant config is the same. Exiting early")
+			os.Exit(0)
```

Comment:
nit: this function would be easier to unit-test if you don't exit(0) here but instead return the decision about early-exit or not and let the main function exit conditionally.

Reply:
Very good point. Will refactor

Reply:
After reviewing, going to revisit this in future PR. There's a larger overhaul to main that I'd like to complete that would work with these type of conditional checks better.